File: software/examples/rust/example_simple.rs
use std::{error::Error, io};

use tinkerforge::{energy_monitor_bricklet::*, ip_connection::IpConnection};

const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Energy Monitor Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let em = EnergyMonitorBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
                                          // Don't use device before ipcon is connected.

    // Get current Energy Data.
    let energy_data = em.get_energy_data().recv()?;

    println!("Voltage: {} V", energy_data.voltage as f32 / 100.0);
    println!("Current: {} A", energy_data.current as f32 / 100.0);
    println!("Energy: {} Wh", energy_data.energy as f32 / 100.0);
    println!("Real Power: {} h", energy_data.real_power as f32 / 100.0);
    println!("Apparent Power: {} VA", energy_data.apparent_power as f32 / 100.0);
    println!("Reactive Power: {} VAR", energy_data.reactive_power as f32 / 100.0);
    println!("Power Factor: {}", energy_data.power_factor as f32 / 1000.0);
    println!("Frequency: {} Hz", energy_data.frequecy as f32 / 100.0);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Energy Monitor Bricklet

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletEnergyMonitor em = new BrickletEnergyMonitor(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current Energy Data
		int voltage, current, energy, realPower, apparentPower, reactivePower,
		    powerFactor, frequency;
		em.GetEnergyData(out voltage, out current, out energy, out realPower,
		                 out apparentPower, out reactivePower, out powerFactor,
		                 out frequency);

		Console.WriteLine("Voltage: " + voltage/100.0 + " V");
		Console.WriteLine("Current: " + current/100.0 + " A");
		Console.WriteLine("Energy: " + energy/100.0 + " Wh");
		Console.WriteLine("Real Power: " + realPower/100.0 + " h");
		Console.WriteLine("Apparent Power: " + apparentPower/100.0 + " VA");
		Console.WriteLine("Reactive Power: " + reactivePower/100.0 + " VAR");
		Console.WriteLine("Power Factor: " + powerFactor/1000.0);
		Console.WriteLine("Frequency: " + frequency/100.0 + " Hz");

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}
